utils/wasm-builder/src/cargo_toolchain.rs

```diff
@@ -35,20 +46,30 @@ impl Toolchain {
         let path = path.into();
 
         // Cargo path format:
-        // "$RUSTUP_HOME/toolchains/**toolchain_name**/bin/cargo"
-        let toolchain_name = path
+        // "$RUSTUP_HOME/toolchains/**toolchain_desc**/bin/cargo"
+        let toolchain_desc = path
             .iter()
             .nth_back(2)
             .and_then(OsStr::to_str)
             .ok_or_else(|| BuilderError::CargoPathInvalid(path.clone()))?;
 
-        // Toolchain name format:
-        // "**toolchain**-arch-arch-arch-arch"
-        let toolchain = toolchain_name
-            .rsplitn(5, '-')
-            .last()
-            .map(String::from)
-            .ok_or_else(|| BuilderError::CargoToolchainInvalid(toolchain_name.into()))?;
+        // This regex is borrowed from the rustup code and modified (added non-capturing groups)
+        lazy_static! {
```

Comment:
Not sure `lazy_static` is good inside the function, but no matters. And I personally prefer `once_cell` as it seems to be in `std` lib in some future 😄 
https://github.com/rust-lang/rust/issues/74465

Reply:
Re-implemented via once_cell